vllm/model_executor/layers/quantization/utils/w8a8_utils.py

```diff
@@ -132,20 +132,17 @@ def apply_fp8_linear(
         per_tensor_weights = (weight_scale.numel() == 1)
         per_tensor_activations = (x_scale.numel() == 1)
 
+        global TORCH_DEVICE_IDENTITY
+        if TORCH_DEVICE_IDENTITY.device != weight.device:
+            TORCH_DEVICE_IDENTITY = TORCH_DEVICE_IDENTITY.to(weight.device)
```

Comment:
This is a really bad code smell that is further evidence for why this should not be initialized as a global, but rather when it is used where the device is already set.